dev/com.ibm.ws.org.apache.cxf.cxf.rt.rs.mp.client.3.2/src/org/apache/cxf/microprofile/client/cdi/CDIInterceptorWrapperImpl.java

```diff
@@ -0,0 +1,168 @@
+            });
+        } catch (PrivilegedActionException pae) {
+            if (LOG.isLoggable(Level.FINEST)) {
+                LOG.log(Level.FINEST, "Exception checking for MP Fault Tolerance class - " +
+                                      "expected if FT feature isnot enabled", pae);
+            }
+            return null;
+        }
+    }
+
+    private static boolean containsFTannotation(Annotation[] interceptorBindings) {
+        for (Annotation anno : interceptorBindings) {
+            String className = anno.annotationType().getName();
+            if (className.startsWith("org.eclipse.microprofile.faulttolerance")) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private static List<Interceptor<?>> getFTInterceptor(BeanManager beanManager) {
+        Annotation ftAnno = new Annotation() {
+
+            @SuppressWarnings("unchecked")
+            @Override
+            public Class<? extends Annotation> annotationType() {
+                return (Class<? extends Annotation>) FT_ANNO_CLASS;
+            }};
+        return beanManager.resolveInterceptors(InterceptionType.AROUND_INVOKE, ftAnno);
+    }
+
+    private static Map<Method, List<InterceptorInvoker>> initInterceptorInvokers(BeanManager beanManager,
+                                                                                 CreationalContext<?> creationalContext,
+                                                                                 Class<?> restClient) {
+        Map<Method, List<InterceptorInvoker>> invokers = new HashMap<>();
+        // Interceptor as a key in a map is not entirely correct (custom interceptors) but should work in most cases
+        Map<Interceptor<?>, Object> interceptorInstances = new HashMap<>();
+
+        List<Annotation> classBindings = getBindings(restClient.getAnnotations(), beanManager);
+
+        for (Method method : restClient.getMethods()) {
+            if (method.isDefault() || Modifier.isStatic(method.getModifiers())) {
+                continue;
+            }
+            List<Annotation> methodBindings = getBindings(method.getAnnotations(), beanManager);
+
+            if (!classBindings.isEmpty() || !methodBindings.isEmpty()) {
+
+                Annotation[] interceptorBindings = merge(methodBindings, classBindings);
+
+                List<Interceptor<?>> interceptors = new ArrayList<>(beanManager.resolveInterceptors(InterceptionType.AROUND_INVOKE, interceptorBindings));
+                if (LOG.isLoggable(Level.FINEST)) {
+                    LOG.finest("Resolved interceptors from beanManager, " + beanManager + ":" + interceptors);
+                }
+                // hack for FT interceptors
+                if (FT_ANNO_CLASS != null && containsFTannotation(interceptorBindings)) {
+                    interceptors.addAll(0, getFTInterceptor(beanManager));
+                }
```

Comment:
Can we add the `FT_ANNO_CLASS` to the `methodBindings` instead? Otherwise the ordering of interceptors will be wrong.

Reply:
Done - good idea, thanks!